plugins/plugin-core-impl/src/connection.ts

```diff
@@ -53,15 +60,21 @@ export async function connect (
   }
   const processor = new RequestProcessor(async () => {
     const s = await doConnect()
-    return {
-      send: (request) => {
-        s.send(serialize(request))
-      }
-    }
-  }, handler)
+    return { send: (req) => s.send(serialize(req)) }
+  })
 
   // Force socket creation.
   doConnect() // eslint-disable-line
 
-  return processor
+  return {
+    async findAll<T extends Doc>(_class: Ref<Class<T>>, query: DocumentQuery<T>): Promise<T[]> {
+      const result = await processor.request('findAll', _class, query)
+      return result as T[]
+    },
+    async tx (tx: Tx): Promise<void> {
+      await processor.request('tx', tx)
+      // Process on server and return result.
+      handler(tx)
+    }
+  }
```

Comment:
Please think to inherit `connection` from `RequestProcessor` and implement `Storage` -- will put many things in shape in this case.

Reply:
Done.